### ISSUE TITLE
Questions regarding VAD fine-tuning.

### ISSUE THREAD

**Issue by gabitza-tech** [NVIDIA/NeMo#5966]
Hello,

I would like to fine-tune the vad multilingual marblenet model using my custom speaker diarization dataset. At the moment, I read the following tutorials:
1. https://colab.research.google.com/github/NVIDIA/NeMo/blob/r1.0.0rc1/tutorials/asr/06_Voice_Activiy_Detection.ipynb#scrollTo=Z7FxGpSE_FyM
2. https://github.com/NVIDIA/NeMo/blob/r1.0.0rc1/tutorials/asr/01_ASR_with_NeMo.ipynb
3. https://github.com/NVIDIA/NeMo/blob/r1.0.0rc1/tutorials/asr/01_ASR_with_NeMo.ipynb

However, I am not sure yet how to adapt these tutorials to the marblenet model. Most of my questions are related to the data preparation and preprocessing step:
1. I read in the marblenet paper that the nn takes a fixed length of 64 dimensional melfilterbank features from 0.63s length segments. Does this mean that I should extract speech and non speech segments of length 0.63s?
2. Should the labels be: "speech" and "non-speech"?
3. Is there any script that takes care of this data preparation and preprocessing part? Or is there any new tutorial similar to the ASR tutorial?
4. Can I find the preprocessing tools such as SpecAugment or SpecCutOut in the nemo toolkit?

Thank you in advance!

**Comment by fayejf** [comment]
Thanks for your questions. 

Finetuning is mainly introduced in Transfer Leaning & Fine-tuning on a new dataset in [Voice_Activity_Detection](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Voice_Activity_Detection.ipynb ) and [Online_Offline_Microphone_VAD_Demo](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Online_Offline_Microphone_VAD_Demo.ipynb) which redirect you to 
 in [ Fine-tuning on a new dataset & Advanced Usage parts in Speech_Commands](https://github.com/NVIDIA/NeMo/blob/stable/tutorials/asr/Speech_Commands.ipynb ) and Transfer learning in [ASR_with_NeMo](https://github.com/NVIDIA/NeMo/blob/stable/tutorials/asr/ASR_with_NeMo.ipynb). 

btw you were looking at very old version of tutorials and you may wanna have a look at updated version. May I know where did you find the r1.0.0rc1 tutorials? We need to update it to new versions. Thanks

> I read in the marblenet paper that the nn takes a fixed length of 64 dimensional melfilterbank features from 0.63s length segments. Does this mean that I should extract speech and non speech segments of length 0.63s?

Yes. [process_vad_data.py#L144](https://github.com/NVIDIA/NeMo/blob/main/scripts/dataset_processing/process_vad_data.py#L144) can help with it. 

> Should the labels be: "speech" and "non-speech"?

 "speech" and "background"

> Is there any script that takes care of this data preparation and preprocessing part? Or is there any new tutorial similar to the ASR tutorial?

You can find how to prepare data in [Voice_Activity_Detection](https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/Voice_Activity_Detection.ipynb) ( I think you have read about it but maybe in older version. )
You can add to yaml file and 
```
init_from_pretrained_model:
  model:
    name: 'vad_multilingual_marblenet'
    include: ["preprocessor","encoder"]
    exclude: ["decoder"] 
```
Specify pre-trained models and use [speech_to_label.py](https://github.com/NVIDIA/NeMo/blob/main/examples/asr/speech_classification/speech_to_label.py) to finetune the model.

` model.maybe_init_from_pretrained_checkpoint(cfg) ` this line will load pretrained model and you can train on your own data. 

> Can I find the preprocessing tools such as SpecAugment or SpecCutOut in the nemo toolkit?

Absolutely, [SpecAugment](https://github.com/NVIDIA/NeMo/blob/5fcfa9e7bf2da39d622071fc0cac592f993f89ea/nemo/collections/asr/parts/submodules/spectr_augment.py#L24) and [SpecCutout](https://github.com/NVIDIA/NeMo/blob/5fcfa9e7bf2da39d622071fc0cac592f993f89ea/nemo/collections/asr/parts/submodules/spectr_augment.py#L108). Just simply add to yaml file like 

```
  spec_augment:
    _target_: nemo.collections.asr.modules.SpectrogramAugmentation
    freq_masks: 2
    time_masks: 2
    freq_width: 15
    time_width: 25
    rect_masks: 5
    rect_time: 25
    rect_freq: 15
```
as in [mabrlenet yaml file]( https://github.com/NVIDIA/NeMo/blob/main/examples/asr/conf/marblenet/marblenet_3x2x64.yaml)

**Comment by gabitza-tech** [comment]
Hi! Thanks a lot for the comprehensive explanation! I understood the process now!
